pkg/space/space.go

```diff
@@ -59,5 +61,14 @@ func NewSubSpace(spaceRequest *toolchainv1alpha1.SpaceRequest, parentSpace *tool
 
 // SubSpaceName generates a name for a subSpace based on parentSpace name and spacerequest UID.
 func SubSpaceName(parentSpace *toolchainv1alpha1.Space, spacerequest *toolchainv1alpha1.SpaceRequest) string {
-	return parentSpace.GetName() + string(spacerequest.UID[:5]) // we only get the first 5 chars from UID so we can keep the name within the length limits
+	parentSpaceName := parentSpace.GetName()
+
+	// if the parent space is itself a subspace, then we need to strip its
+	// identifier out to prevent length limitations from kicking in
+	if parentSpace.Spec.ParentSpace != "" {
```

Comment:
Let's also check that the parent space name is actually longer than 6 characters. Just to be 100% on the safe side.

Reply:
Fixed in 46b3f1e.